### PR TITLE
Made domain and status calls run in parallel

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #4910

## Description of Changes
- Title

## Test Plan
- Open homepage, check that domain and status calls timings overlap in chrome debugger tools
- Let CI suite run
- Start ssl proxy to test custom domain. Made sure that it navigated to that communities homepage

### CODE REVIEW

**Comment by jnaviask** [comment]
@kurtisassad infinite cow bobber on custom domains for me, vs master where custom domains load fine.

**Comment by kurtisassad** [comment]
Good catch. Fixed

Wondering if it would be worthwhile adding custom domain tests to the e2e suite (Just a basic loading page)

If so, mention me and I will add a ticket.

**Comment by jnaviask** [comment]
@kurtisassad fantastic idea -- feel free to ticket it up

**Comment by kurtisassad** [comment]
Ok fixed now

**Comment by mzparacha** [comment]
LGTM @jnaviask

File: packages/commonwealth/client/scripts/hooks/useInitApp.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
-    fetch('/api/domain')
-      .then((res) => res.json())
+    Promise.all([axios.get(`${app.serverUrl()}/domain`), initAppState()])
+      .then(([domainResp]) => {
+        return domainResp.data;
+      })
       .then(({ customDomain: serverCustomDomain }) => {
+        if (serverCustomDomain) {
+          app.setCustomDomain(serverCustomDomain);
+        }
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/hooks/useInitApp.ts:17]: can we use a single then here?

Comment (jnaviask) [reply packages/commonwealth/client/scripts/hooks/useInitApp.ts:17]: cc @kurtisassad

**Comment by kurtisassad** [comment]
Re-tested with testplan after change. Works fine

**Pull request merged**